services/discordService.js

```diff
@@ -30,6 +33,26 @@ const getDiscordMembers = async () => {
   }
 };
 
+const setInDiscordFalseScript = async () => {
+  const users = await fetchAllUsers();
+  const updateUsersPromises = [];
+  users.forEach((user) => {
+    const id = user.id;
```

Comment:
delete user[id] statement deletes the id property from the user object. This means that the id property will no longer be present in the userData object that is passed to the userModel.doc(id).update(userData) method. This could allow an attacker to inject malicious code into the database by creating a user object with an id property that contains malicious code.

can't reframe like below
const setInDiscordFalseScript = async () => {
  const users = await fetchAllUsers();
  const updateUsersPromises = [];
  users.forEach((user) => {
    const userData = Object.assign({}, user, {
      roles: {
        ...user.roles,
        in_discord: false,
      },
    });
    updateUsersPromises.push(userModel.doc(user.id).update(userData));
  });
  await Promise.all(updateUsersPromises);
};